R/bundle.R

```diff
@@ -1119,3 +1135,59 @@ performPackratSnapshot <- function(bundleDir, verbose = FALSE) {
   # TRUE just to indicate success
   TRUE
 }
+
+# Run "quarto inspect" on the target and returns its output as a parsed object.
+quartoInspect <- function(appDir = NULL, appPrimaryDoc = NULL, quarto = NULL) {
+  if (!is.null(quarto)) {
+    inspect <- NULL
+    # If "quarto inspect appDir" fails, we will try "quarto inspect
+    # appPrimaryDoc", so that we can support single files as well as projects.
+    primaryDocPath <- file.path(appDir, appPrimaryDoc) # prior art: appHasParameters()
+    for (path in c(appDir, primaryDocPath)) {
+      args <- c("inspect", path.expand(path))
+      inspect <- suppressWarnings(system2(quarto, args, stdout = TRUE, stderr = FALSE))
+      if (is.null(attr(inspect, "status"))) {
+        return(jsonlite::fromJSON(inspect))
+      }
+    }
+  }
+  return(NULL)
+}
+
+# Extract the Quarto version and engines, either from a parsed "quarto inspect"
+# result or from metadata provided by the quarto-r package.
+getQuartoManifestDetails <- function(inspect = list(), metadata = list()) {
```

Comment:
nit: I would prefer if this were split into two "constructor" functions, rather than one function that consumed the inputs differently depending upon what is present. Frankly, it might be simpler if the behavior were just inlined into inferQuartoInfo, but I don't feel strongly.

Reply:
I agree — I'll inline it.

Reply:
Agreed. I'll inline it. I think that having two separate `list()` statements in two separate functions is a bit too much indirection.